devel/boost171/Portfile

```diff
@@ -0,0 +1,483 @@
+# require ports depending on Boost to also require C++11 compliance,
+# and requiring it does make such building easier for those ports.
+configure.cxxflags-append -std=gnu++11
+compiler.cxx_standard     2011
+
+# It turns out that ccache and distcc can produce boost libraries that, although they
+# compile without warning, have all sorts of runtime errors especially with pointer corruption.
+# Since most people will now use MacPorts' pre-compiled boost, this should not be a problem.
+configure.ccache    no
+configure.distcc    no
+
+configure.cmd       ./bootstrap.sh
+configure.args      --without-libraries=python \
+                    --without-libraries=mpi \
+                    --with-icu=${prefix}
+
+# boost build scripts default to clang on darwin
+if {[string match *gcc* ${configure.compiler}]} {
+    configure.args-append --with-toolset=gcc
+}
+
+if {${os.platform} eq "darwin" && ${os.major} <= 10} {
+    configure.args-append   --without-libraries=context \
+                            --without-libraries=coroutine
+}
+
+configure.universal_args
+
+post-configure {
+
+    reinplace -E "s|-install_name \"|&${bprefix}/lib/|" \
+        ${worksrcpath}/tools/build/src/tools/darwin.jam
+
+    set compileflags ""
+    if {[string length ${configure.sdkroot}] != 0} {
+        set compileflags "<compileflags>\"-isysroot ${configure.sdkroot}\""
+    }
+
+    set cxx_stdlibflags {}
+    if {[string match *clang* ${configure.cxx}]} {
+        set cxx_stdlibflags -stdlib=${configure.cxx_stdlib}
+    }
+
+    # see https://trac.macports.org/ticket/55857
+    # see https://svn.boost.org/trac10/ticket/13454
+    write_jam "using darwin : : ${configure.cxx} : <asmflags>\"${configure.cflags} [get_canonical_archflags cc]\" <cflags>\"${configure.cflags} [get_canonical_archflags cc]\" <cxxflags>\"${configure.cxxflags} [get_canonical_archflags cxx] ${cxx_stdlibflags}\" ${compileflags} <linkflags>\"${configure.ldflags} ${cxx_stdlibflags}\" : ;"
+
+}
+
+build.cmd       ${worksrcpath}/b2
+build.target
+build.args      -d2 \
+                --layout=tagged \
+                --debug-configuration \
+                --user-config=user-config.jam \
+                -sBZIP2_INCLUDE=${prefix}/include \
+                -sBZIP2_LIBPATH=${prefix}/lib \
+                -sEXPAT_INCLUDE=${prefix}/include \
+                -sEXPAT_LIBPATH=${prefix}/lib \
+                -sZLIB_INCLUDE=${prefix}/include \
+                -sZLIB_LIBPATH=${prefix}/lib \
+                -sICU_PATH=${prefix} \
+                variant=release \
+                threading=single,multi \
+                link=static,shared \
+                runtime-link=shared \
+                -j${build.jobs} \
+                --no-cmake-config
+
+destroot.cmd    ${worksrcpath}/b2
+destroot.post_args
+
+pre-destroot {
+    destroot.args {*}${build.args} --prefix=${destroot}${bprefix}
+    system "find ${worksrcpath} -type f -name '*.gch' -exec rm {} \\;"
+}
+
+proc boost_docs_install {} {
```

Comment:
can this `proc` be moved into the boost PG instead of duplicating it in every individual Portfile?

Reply:
No, because the boost ports do not themselves use the boost PG

Reply:
What would help here is if the boost ports are merged into a single port file with subports for each version. This was previously discussed but is up to the maintainers to do if they wish in a future update.